sdk/core/azure-core/azure/core/pipeline/policies/_sensitive_header_cleanup_policy.py

```diff
@@ -0,0 +1,72 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+from typing import List, Optional, Any, TYPE_CHECKING
+from ._base import SansIOHTTPPolicy
+
+if TYPE_CHECKING:
+    from azure.core.pipeline import PipelineRequest
+
+
+class SensitiveHeaderCleanupPolicy(SansIOHTTPPolicy):
+    """A simple policy that cleans up sensitive headers
+
+    :keyword List[str] block_headers_list: The headers to clean up.
+    :keyword bool disable_cleanup: Opt out cleaning up sensitive headers
+    """
+
+    DEFAULT_SENSITIVE_HEADERS = set(
+        [
+            "Authorization",
+            "x-ms-authorization-auxiliary",
+        ]
+    )
+
+    def __init__(
+        self,  # pylint: disable=unused-argument
+        *,
+        block_headers_list: Optional[List[str]] = None,
+        disable_cleanup: bool = False,
```

Comment:
These two parameters will be configured as part of the pipeline configuration, and therefore bundled into the single `kwargs` that is passed into the PipelineClient configuration (including possibly set from customer kwargs at client creation). Therefore:
- Their names probably need to be unique to the pipeline, e.g.:
    - `disable_cleanup` to maybe `redirect_disable_cleanup` or `disable_redirect_cleanup`
    - `block_headers_list` to maybe `redirect_block_headers` or `blocked_redirect_headers`
    - (using the policy as the arg prefix was an early convention using in some policies, so may or may not be useful here)
- The new params should be added to the pipeline policy configuration table in the dev docs (along with any other parts of that doc that may need to be updated to include mention of this policy):
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md#policies